### PR TITLE
[fix][broker] Fix loadbalance score caculation problem

### PR DESCRIPTION
**Pull request by gaozhangmin**
### Motivation
There is a problem in my environment, a bundle cannot be loaded to an empty broker(no bundles on it). 
I found that this empty broker still has preallocatedBundleData on it which caused the score is higher than others.

https://github.com/apache/pulsar/blob/c890a8c15cd9d30518818377c3155776514f5a9c/pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/LeastLongTermMessageRate.java#L53-L78

This issue is caused by the preallocatedBundleData is not cleared from brokerData. LINE 573 and LINE 580 is the condition  of clearing preallocatedBundleData.  

There is a corner case, which bundle is unload before broker data reportted to Zookeeper.  the preallocatedBundleData  would never have the chance of clearing from loadData of leader.

https://github.com/apache/pulsar/blob/c890a8c15cd9d30518818377c3155776514f5a9c/pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java#L570-L590

### Modifications

If current broker doesn't own the bundle, clearing the preallocatedBundleData

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository
https://github.com/gaozhangmin/pulsar/pull/10

### CODE REVIEW

**Review by HQebupt — APPROVED** [review]
LGTM

**Review by jiazhai — APPROVED** [review]

**Review by Technoboy- — APPROVED** [review]

**Comment by lhotari** [comment]
If there was a bug, why didn't we try to add a unit test to reproduce the issue and prevent further regressions?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java [hunk 1]
```diff
@@ -568,24 +570,19 @@ private void updateBundleData() {
...
-                                        .iterator();
-                        while (preallocatedIterator.hasNext()) {
-                            final String bundle = preallocatedIterator.next().getKey();
-
-                            if (bundleData.containsKey(bundle)) {
-                                preallocatedIterator.remove();
-                                preallocatedBundleToBroker.remove(bundle);
-                            }
-                        }
+                preallocatedBundleToBroker.keySet().removeAll(preallocatedBundleData.keySet());
```
Comment (lhotari) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java:576]: `keySet().removeAll` seems wrong

Comment (lhotari) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java:576]: Oh it seems that it's supported, at least for `HashMap.keySet()`.

**Pull request merged**